server/src/test/java/io/crate/expression/reference/sys/snapshot/SysSnapshotRestoreTest.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.expression.reference.sys.snapshot;
+
+import io.crate.test.integration.CrateDummyClusterServiceUnitTest;
+import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.cluster.RestoreInProgress;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.collect.ImmutableOpenMap;
+import org.elasticsearch.index.shard.ShardId;
+import org.elasticsearch.snapshots.Snapshot;
+import org.elasticsearch.snapshots.SnapshotId;
+import org.junit.Test;
+
+import java.util.List;
+import java.util.UUID;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class SysSnapshotRestoreTest extends CrateDummyClusterServiceUnitTest {
+
+    @Test
+    public void test_extract_and_convert_one_restore_in_progress_to_sys_restore_snapshot_context_type() {
+        ImmutableOpenMap.Builder<ShardId, RestoreInProgress.ShardRestoreStatus> shardsBuilder
+            = ImmutableOpenMap.builder();
+        shardsBuilder.put(
+            new ShardId("index", "_uuid", 0),
+            new RestoreInProgress.ShardRestoreStatus("nodeId", RestoreInProgress.State.STARTED)
+        );
+        RestoreInProgress.Entry entry = new RestoreInProgress.Entry(
+            "_uuid",
+            new Snapshot(
+                "repository",
+                new SnapshotId("snapshot", UUID.randomUUID().toString())),
+            RestoreInProgress.State.SUCCESS,
+            List.of("index"),
+            shardsBuilder.build()
+        );
+
+        var clusterService = mock(ClusterService.class);
+        var clusterState = mock(ClusterState.class);
+        when(clusterState.custom(RestoreInProgress.TYPE))
+            .thenReturn(new RestoreInProgress.Builder().add(entry).build());
+        when(clusterService.state()).thenReturn(clusterState);
```

Comment:
I'm not sure if it is worth testing on this level if the clusterService and state is mocked.

How about changing the `snapshotsRestoreInProgress` method to take the `ClusterState` or even the `RestoreInProgress` as an argument? I think it could also move into the new `SysSnapshotRestoreTableInfo` class as a static method.


`SysTableDefinitions` could then directly call that method - it already has a clusterService instance - and the new `SysSnapshotsRestore` class could be removed again.

Reply:
Pushed the fixup. I was also not sure about `SysSnapshotRestore`..